deploy/deployUtils.ts

```diff
@@ -226,24 +458,104 @@ export async function checkTokens(
-        { from: deployer, log: true },
-        "mint",
-        deployer,
-        BigNumber.from(10).pow(decimals).mul(1000000),
-      )
+      for (const token in tokenArgs) {
+        await deploy(token, {
+          from: deployer,
+          log: true,
+          contract: "GenericERC20",
+          args: tokenArgs[token],
+          skipIfAlreadyDeployed: true,
+        })
+        const decimals = tokenArgs[token][2]
+        console.log(`minting: ${tokenArgs[token][0]}`)
+        await execute(
+          token,
+          { from: deployer, log: true },
+          "mint",
+          deployer,
+          BigNumber.from(10).pow(decimals).mul(1000000),
+        )
+      }
     }
   }
 }
+
+async function checkRegisteredPool(
+  hre: HardhatRuntimeEnvironment,
+  poolName: string,
+) {
+  const { ethers, deployments } = hre
+  const { log } = deployments
+
+  const poolRegistry: PoolRegistry = await ethers.getContract("PoolRegistry")
+
+  await poolRegistry
+    .getPoolDataByName(poolName)
+    .then(() => {
+      log("Skipping adding pools to registry because they are already added")
+      return false
+    })
+    .catch(async () => {
+      return true
+    })
+}
+
+export async function registerPools(
+  hre: HardhatRuntimeEnvironment,
+  pools: IPoolDataInput[],
+) {
+  const { deployments, getNamedAccounts, ethers } = hre
+  const { execute, get } = deployments
+  const { deployer } = await getNamedAccounts()
+  const poolRegistry: PoolRegistry = await ethers.getContract("PoolRegistry")
+  const poolsToBeAdded = pools.filter((pool) =>
+    checkRegisteredPool(hre, pool.poolName.toString()),
+  )
+  console.log(`Attempting to register ${poolsToBeAdded.length} pool[s]`)
+  let poolsToBeRegistered: IPoolRegistry.PoolInputDataStruct[] = []
+  poolsToBeAdded.forEach(async (pool) => {
+    poolsToBeRegistered.push({
+      poolAddress: (await get(pool.poolName)).address,
+      typeOfAsset: PoolType.USD,
+      poolName: ethers.utils.formatBytes32String(pool.tokenNames.join("-")),
+      targetAddress: (await get("SwapFlashLoan")).address,
+      metaSwapDepositAddress: ZERO_ADDRESS,
+      isSaddleApproved: true,
+      isRemoved: false,
+      isGuarded: false,
+    })
+  })
+
+  const batchCall = await Promise.all(
+    poolsToBeRegistered.map(
+      async (pool) => await poolRegistry.populateTransaction.addPool(pool),
+    ),
+  )
+  const batchCallData = batchCall.map((x) => x.data).filter(Boolean)
+
+  await execute(
+    "PoolRegistry",
+    { from: deployer, log: true },
+    "batch",
+    batchCallData,
+    true,
+  )
+}
+
+export async function verifyContract(
+  hre: HardhatRuntimeEnvironment,
+  contractName: string,
+  constructors: any[] = [],
+) {
+  const { ethers } = hre
+  const contract = await ethers.getContract(contractName)
+
+  console.log(`attempting to verify contract: ${contractName}`)
+  await hre.run("verify:verify", {
+    address: contract.address,
+    constructorArguments: constructors,
+  })
```

Comment:
Note that this is using `etherscan-verify`, not `hardhat-deploy`

Therefore will not work on all networks.